tidy3d/components/structure.py

```diff
@@ -92,6 +93,19 @@ def eps_diagonal(self, frequency: float) -> Tuple[complex, complex, complex]:
         """
         return self.medium.eps_diagonal(frequency=frequency)
 
+    @pydantic.validator("medium", always=True)
+    def _check_2d_geometry(cls, val, values):
+        """Medium2D is only consistent with certain geometry types"""
+        geom = values["geometry"]
+        if isinstance(val, Medium2D):
+            if not isinstance(geom, Box):
+                raise ValidationError("Medium2D is only compatible with Box geometry.")
```

Comment:
put single quotes `'` around variable or class names in the exception messages for better terminal readability.

Reply:
and below